Source/ACE/Entity/Landblock.cs

```diff
@@ -509,6 +510,25 @@ public void EnqueueBroadcast(Position pos, float distance, params GameMessage[]
         }
 
         /// <summary>
+        /// Enqueues a message for global broadcast, thread safe
+        /// </summary>
+        /// <param name="msg"></param>
+        public void EnqueueGlobalBroadcast(params GameMessage[] msgs)
+        {
+            // Atomically checks and sets the broadcastQueued bit --
+            //    guarantees that if we need a broadcast it will be enqueued in the world-managers broadcast queue exactly once
+            if (Interlocked.CompareExchange(ref broadcastQueued, 1, 0) == 0)
+            {
+                WorldManager.BroadcastQueue.EnqueueAction(new ActionEventDelegate(() => SendGlobalBroadcasts()));
```

Comment:
The reason that EnqueueBroadcasts has a "BroadcastQueue" is because it must wait until it knows the positions of objects in the world are not changing before doing its position-based broadcast checks, otherwise, it will race when trying to send position based broadcasts.

As you said your doing a broadcast to all things in the world, regardless of position.  You don't have the requirement which dictates the work-queue, yet you still chose to queue up the operation to send later.  This decision doesn't make sense to me.   Why don't you just make a "BroadcastToAllPlayers" function in whatever code manages our active sessions (grabbing a lock on this shouldn't be a big deal, as it s rarely changed), or do we not maintain a list of active sessions for some reason?